extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/UndertowBuildStep.java

```diff
@@ -153,6 +148,7 @@ public ServiceStartBuildItem boot(UndertowDeploymentTemplate template,
             LaunchModeBuildItem launchMode,
             ExecutorBuildItem executorBuildItem,
             HttpConfig config) throws Exception {
+        CORSTemplate.httpConfig = config;
```

Comment:
This does not work, you need to create a non-static method on the CORSTemplate, and pass the config into that method (Confusingly thought this will work in tests, as tests are run on the same JVM, but in production the build step code happens in a different JVM to the execution of the template, I probably should try and use an isolated class loader in tests to catch these errors).

 Other than that I think this is ok. I think we need some general improvements in our core config code to handle situations like this, however I am not 100% sure what they should be.

Reply:
Cool, I'm going to change this and push the changes.